pkg/validation/errors.go

```diff
@@ -161,6 +161,36 @@ func (e *InvalidProjectCheckoutRemoteError) Error() string {
 	return fmt.Sprintf("unable to find the checkout remote %s in the remotes for %s %s", e.checkoutRemote, e.objectType, e.objectName)
 }
 
+type ResourceRequirementType string
+
+const (
+	MemoryLimit   ResourceRequirementType = "memoryLimit"
+	CpuLimit      ResourceRequirementType = "cpuLimit"
+	MemoryRequest ResourceRequirementType = "memoryRequest"
+	CpuRequest    ResourceRequirementType = "cpuRequest"
```

Comment:
I would suggest to just use the const from `core/v1` but its up to you

Reply:
I don't want to involve core/v1 in for the devfile validation, the validation should only for devfile. 
especially the const in core/v1 are `limits.cpu` and `limits.memory`, the values are different from devfile variable names